lnrpc/lightning.proto

```diff
@@ -619,6 +631,50 @@ message LookupHtlcResolutionResponse {
     bool offchain = 2;
 }
 
+// UpdateChannelParamsRequest defines a request to be used when upgrading the
+// channel. All its params are optional except the channel point. In addition,
+// at least one optional field must be set to make it a valid request.
+message UpdateChannelParamsRequest {
+    ChannelPoint channel_point = 1;
+
+    // The dust limit of the initiator's commitment tx.
+    optional uint64 dust_limit = 2;
+
+    // The maximum amount of coins in millisatoshis that can be pending in this
+    // channel.
+    optional uint64 max_value_in_flight = 3;
+
+    // The minimum amount of satoshis the initiator requires us to have at all
+    // times.
+    optional uint64 channel_reserve = 4;
+
+    // The smallest HTLC in millisatoshis that the initiator will accept.
+    optional uint64 min_htlc = 5;
+
+    /*
+    The number of blocks to use for the relative time lock in the
+    pay-to-self output of both commitment transactions.
+    */
+    optional uint32 csv_delay = 6;
```

Comment:
There's a bit of nuance here, as some of these fields are actually _asymmetric_ when applied to a channel. 

As an example, the `csv_delay` we send in `open_channel` is actually what we want _them_ to use for _their_ commitment transaction. 

I think we can hide this fact from the user with this RPC, but it's something to keep in mind during the "negotiation" phase. As if we want to reduce _our_ csv delay, we need to ask the remote peer if their willing to. This is security sensitive, as if the remote party just always approvals this subset of params, then I can get them to reduce it to 1 block, and try to breach (they have less time to react). 

That's all to say that certain params can be auto accept, while others may need som edefalut setting or a sort of acceptor subscription.